### PR TITLE
Quotes properly using goog.string.quote instead of the hacky regex

### PR DESCRIPTION
**Pull request by Goom11**
This will make it easier for whomever implements multi-line string blocks in the future.

### CODE REVIEW

**Comment by Goom11** [comment]
I signed it!

File: generators/dart.js [hunk 1]
```diff
@@ -149,12 +149,7 @@ Blockly.Dart.scrubNakedValue = function(line) {
  * @private
  */
 Blockly.Dart.quote_ = function(string) {
-  // TODO: This is a quick hack.  Replace with goog.string.quote
-  string = string.replace(/\\/g, '\\\\')
-                 .replace(/\n/g, '\\\n')
-                 .replace(/\$/g, '\\$')
```
Comment (NeilFraser) [inline generators/dart.js]: goog.string.quote does not escape $ (needed for Dart strings).

File: generators/javascript.js [hunk 2]
```diff
@@ -169,11 +169,7 @@ Blockly.JavaScript.scrubNakedValue = function(line) {
  * @private
  */
 Blockly.JavaScript.quote_ = function(string) {
-  // TODO: This is a quick hack.  Replace with goog.string.quote
-  string = string.replace(/\\/g, '\\\\')
-                 .replace(/\n/g, '\\\n')
-                 .replace(/'/g, '\\\'');
-  return '\'' + string + '\'';
+  return goog.string.quote(string);
```
Comment (NeilFraser) [inline generators/javascript.js]: The Google style guide (that we try to follow) recommends using single-quoted string literals in JavaScript:
https://google.github.io/styleguide/javascriptguide.xml?showone=Strings#Strings
goog.string.quote uses double-quotes.  Same issue applies to Dart.

File: generators/python.js [hunk 3]
```diff
@@ -152,12 +152,7 @@ Blockly.Python.scrubNakedValue = function(line) {
  * @private
  */
 Blockly.Python.quote_ = function(string) {
-  // TODO: This is a quick hack.  Replace with goog.string.quote
-  string = string.replace(/\\/g, '\\\\')
-                 .replace(/\n/g, '\\\n')
-                 .replace(/\%/g, '\\%')
```
Comment (NeilFraser) [inline generators/python.js]: goog.string.quote does not escape % (needed for Python strings).

**Comment by NeilFraser** [comment]
I've modified the comments so that they don't mislead developers into thinking that goog.string.quote will work (which is what I thought when I first wrote that comment).
https://github.com/google/blockly/commit/bfa842c9a8396052a8cb2a365ca2270f1c2bc0e8

**Pull request closed without merging**